### DISCUSSION TITLE
prisma upsert returns correct results, but db field is empty

### DISCUSSION THREAD

**Discussion by fstrauf** [prisma/prisma#19738] [Q&A] [answered]
Hi experts,

I'm updating my planetscale db after a stripe subscription calls my webhook:
```js
const response = await prisma.subscriptions.upsert({
    where: {
      authorClerkId: userId,
    },
    create: {
      authorClerkId: userId,
      stripeCustomerId: customer,
      tier: productTier,
    },
    update: {
      stripeCustomerId: customer,
      tier: productTier,
    },
  })
```
            
The response I see in the vercel logs looks fine:

```js
response: {
  authorClerkId: 'user_42',
  stripeCustomerId: 'cus_42',
  tier: 'prod_42',
  exampleSectionCounter: null,
  exampleViewStart: null
}
```

**However, when I check my database, the tier field is empty**. Here is the schema definition:
```prisma
model Subscriptions {
  authorClerkId         String    @id
  stripeCustomerId      String    @unique @default(cuid())
  tier                  String
  exampleSectionCounter Int?
  exampleViewStart      DateTime?
}
```

I can't reproduce the issue if I'm running the stripe webhook in test mode on localhost and I can't reproduce it if I resend the webhook on prod. It only occurs if the webhook gets triggered the first time (i.e. when a client subscribes).

Do you have any suggestions on how to look into this more? Where do things go wrong?

Thanks!

**Comment by ludralph** [comment]
Hi @fstrauf 👋 

Can you please enable [debugging](https://www.prisma.io/docs/concepts/components/prisma-client/debugging) and share the logs here? We should be able to get more information from there.

**Reply by ludralph** [reply]
Hey @fstrauf ,

it would be really helpful if you could provide the additional information that we had asked about last time. If your problem is resolved by now or somehow doesn't feel relevant any more, you can simply close the Discussion as well.

**Reply by fstrauf** [reply, marked as answer]
Thanks for the reminder and apologies for leaving this out here.

The issue was caused by two webhooks being triggered and working against the same database (I had one for preview and one for prod that both got triggered by a new purchase).

These both called the database and this presumably caused the issue.
